### PR TITLE
MNT: Make 'stop' an optional piece of the API.

### PR DESCRIPTION
**Pull request by danielballan**
To discuss before merging.

### CODE REVIEW

File: bluesky/run_engine.py [hunk 1]
```diff
@@ -1730,3 +1724,15 @@ def _default_md_validator(md):
             "GOOD: sample='dirt' "
             "GOOD: sample={'color': 'red', 'number': 5} "
             "BAD: sample=[1, 2] ")
+
+def _try_stopping(obj):
```
Comment (tacaswell) [inline bluesky/run_engine.py]: can we call this `_stop_me_maybe`?

**Comment by tacaswell** [comment]
This seems reasonable :+1:

File: bluesky/run_engine.py (hunk 1 above)
Comment (klauer) [inline bluesky/run_engine.py]: Oh, @tacaswell...

File: bluesky/run_engine.py [hunk 2]
```diff
@@ -1730,3 +1724,15 @@ def _default_md_validator(md):
...
+    "Call obj.stop() and log any exceptions."
+    try:
+        stop = obj.stop
+    except AttributeError:
+        logger.debug("No 'stop' method available on %r", obj)
+    else:
+        try:
+            stop()
+        except Exception:
+            logger.error("Failed to stop %r", obj)
```
Comment (klauer) [inline bluesky/run_engine.py]: How do we feel about the lack of `exc_info` here?

Comment (tacaswell) [inline bluesky/run_engine.py]: it's no worse that it was before.

File: bluesky/run_engine.py [hunk 3]
```diff
@@ -439,10 +439,7 @@ def request_pause(self, defer=False, name=None, callback=None):
             obj.clear_sub(cb)
         # During pause, all motors should be stopped.
         for obj in self._movable_objs_touched:
-            try:
-                obj.stop()
-            except Exception:
-                logger.error("Failed to stop %r", obj)
+            _try_stopping(obj)
```
Comment (tacaswell) [inline bluesky/run_engine.py]: excetpions in `stop` here probably should kill the scan

File: bluesky/run_engine.py (hunk 2 above)
Comment (klauer) [inline bluesky/run_engine.py]: That doesn't really answer the question though - shouldn't we know why it failed to stop?

**Comment by danielballan** [comment]
Force-pushed changes:

* Made this a private method so it can use the RunEngine's `log` attribute.
* Included exception in log message, per @klauer's comment.

**Pull request merged**